elfi/distributions.py

```diff
@@ -32,6 +32,10 @@ def pdf(self, x, *params, **kwargs):
     def logpdf(self, x, *params, **kwargs):
         raise NotImplementedError
 
+    @property
+    def name(self):
+        return self._name or self.__class__.__name__
```

Comment:
`self.__class__.__name__` is not needed here since it's being set in the constructor.